brushfire-core/src/main/scala/com/stripe/brushfire/Tree.scala

```diff
@@ -19,37 +19,31 @@ object Tree {
           if(sampler.includeFeature(f, treeIndex, leaf.index))
             Map(f -> splitter.create(v, instance.target))
           else
-            Map.empty[K,splitter.S]
+            Map.empty[K, splitter.S]
         }
       }).flatMap { featureMap =>
-        val splits = featureMap.toList.flatMap {
-          case (f, s) =>
-            splitter.split(leaf.target, s).map { x => f -> evaluator.evaluate(x) }
-        }
 
-        if(splits.isEmpty)
-          None
-        else {
-          val (splitFeature, (split, _)) = splits.maxBy { case (f, (x, s)) => s }
-          val edges = split.predicates.toList.map {
-            case (pred, _) =>
-              val newInstances = instances.filter { inst => pred(inst.features.get(splitFeature)) }
-              val target = Monoid.sum(newInstances.map { _.target })
-              (pred, target, newInstances)
-          }
-
-          if (edges.count { case (_, _, newInstances) => newInstances.nonEmpty } > 1) {
-            Some(SplitNode(edges.map {
-              case (pred, target, newInstances) =>
-                (splitFeature, pred, expand[K, V, T](times - 1, treeIndex, LeafNode(0, target), splitter, evaluator, stopper, sampler, newInstances))
-            }))
-          } else {
-            None
+        val splits = for {
+          (f, s) <- featureMap.toList
+          split <- splitter.split(leaf.target, s)
+          tpl <- evaluator.evaluate(split)
+        } yield (f, tpl)
+
+        if (splits.isEmpty) None else {
+          val (splitFeature, (Split(pred, left, right), _)) = splits.maxBy { case (f, (x, s)) => s }
+          def ex(dist: T): Node[K, V, T, Unit] = {
```

Comment:
I think it's just meant to abbreviate "expand" which is what we're doing to the right and left children. It's basically just some pre-processing before calling `expand` recursively.